### PR TITLE
Enhancements: Animated GitHub Activity, Responsive Contributor Section, Improved Design Elements 

### LINKED ISSUE #288: Improve mobile responsiveness of Contributors page
image

### PR DESCRIPTION
**Pull request by Lobi29**
<h1 align="center">fixes #288</h1>

### The PR has been updated with the previous recommendation:

- Changed the gray to secondary.
- Moved the motion.div to the parent component.

### CODE REVIEW

File: components/contributors/GithubActivity.tsx [hunk 1]
```diff
@@ -359,51 +380,67 @@ export default function GithubActivity({ activityData }: Props) {
...
-          <option value="last-month">Last 30 days</option>
-          {rangePresets?.map((preset) => (
-            <option key={preset} value={preset}>
-              {new Date(preset).toLocaleString("default", {
-                month: "long",
-                year: "numeric",
-              })}
+    <div className="flex flex-col justify-center md:flex-row  md:justify-between ">
+      <div className="mr-10 w-full md:w-2/5">
+        <div className="top-20 my-4 flex flex-col gap-2 rounded p-4 shadow-md dark:bg-primary-900 md:sticky">
```
Comment (rithviknishad) [inline components/contributors/GithubActivity.tsx:385]: IMO the background with primary color does not go well as we don't seem to use primary shade for background.

image

```suggestion
        <div className="top-20 my-4 flex flex-col gap-2 rounded p-4 shadow-md dark:bg-secondary-800 md:sticky">
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]
The animation makes it feel the page is less responsive. Could we make the animation start faster with lesser delay?

**Comment by Lobi29** [comment]
Hey @rithviknishad,

I've cut down the delay of motion.div to just 0.2 seconds and switched the background color to our secondary base

File: tailwind.config.js [hunk 2]
```diff
@@ -76,5 +76,8 @@ module.exports = {
       },
     },
   },
-  plugins: [require("@tailwindcss/typography")],
+  plugins: [
+    require("tailwind-scrollbar")({ nocompatible: true }),
```
Comment (rithviknishad) [inline tailwind.config.js:80]: why was this added?

**Review by rithviknishad — CHANGES_REQUESTED** [review]
1. Would it be possible to not restart the animation when we scroll back up?
2. When someone wants to take a quick look over what the contributor has been up to, the animation makes it feel slow. A delay of 0.05 would be ideal maybe?

File: tailwind.config.js (hunk 2 above)
Comment (Lobi29) [reply tailwind.config.js:80]: > why was this added?

Since we are using a custom scrollbar in the select options field.

**Comment by Lobi29** [comment]
> 1. Would it be possible to not restart the animation when we scroll back up?
> 2. When someone wants to take a quick look over what the contributor has been up to, the animation makes it feel slow. A delay of 0.05 would be ideal maybe?



Yes, it's possible. Using the useInView feature from framer-motion, we can implement that. However, since my exams are ongoing until March 14th, I am unable to work on this right now. 😔

**Comment by rithviknishad** [comment]
Closing as stale due to no activity + conflicts + not a requirement. Let's keep it minimal.
